scripts/skills/effects/rf_sapling_harvest.nut

```diff
@@ -0,0 +1,45 @@
+this.rf_sapling_harvest <- ::inherit("scripts/skills/skill", {
+	m = {
+		DamageThreshold = 0.1
+	},
+	function create()
+	{
+		this.m.ID = "effects.rf_sapling_harvest";
+		this.m.Name = "Sapling Harvest";
+		this.m.Description = "Whenever this character receives 10% or more damage from a single hit, spawn a Sapling on an adjacent Tile, if possible.";
+		this.m.Icon = "skills/terrain_icon_06.png";
+		this.m.SoundOnUse = [];
+		this.m.Type = this.Const.SkillType.StatusEffect;
+		this.m.Order = this.Const.SkillOrder.Last;
+		this.m.IsActive = false;
+		this.m.IsStacking = false;
+		this.m.IsHidden = false;
+	}
+
+	function onDamageReceived( _attacker, _damageHitpoints, _damageArmor )
+	{
+		local actor = this.getContainer().getActor();
+		if (_damageHitpoints >= actor.getHitpointsMax() * this.m.DamageThreshold)
+		{
+			local candidates = [];
+			local myTile = actor.getTile();
+
+			for( local i = 0; i < 6; i = ++i )
```

Comment:
Two changes for convention's sake:
1. spacing: `for (local i =0; i < 6; i++)`
2. The last part is just `++i` or `i++` and not `i = ++i`